java/src/jni/h5util.c

```diff
@@ -3634,14 +3783,64 @@ Java_hdf_hdf5lib_H5_H5export_1dataset(JNIEnv *env, jclass clss, jstring file_exp
         UNPIN_JAVA_STRING(ENVONLY, file_export_name, file_export);
     if (object_name)
         UNPIN_JAVA_STRING(ENVONLY, object_path, object_name);
-    if (fileName)
-        UNPIN_JAVA_STRING(ENVONLY, file_name, fileName);
     if (dataset_id >= 0)
         H5Dclose(dataset_id);
-    if (file_id >= 0)
-        H5Fclose(file_id);
 } /* end Java_hdf_hdf5lib_H5_H5export_1dataset */
 
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5export_attribute
+ * Signature: (Ljava/lang/String;JLjava/lang/String;I)V
+ */
+JNIEXPORT void JNICALL
+Java_hdf_hdf5lib_H5_H5export_1attribute(JNIEnv *env, jclass clss, jstring file_export_name, jlong dset_id,
+                                        jstring object_path, jint binary_order)
```

Comment:
Would probably make more sense to call this "attribute_name" rather than "object_path", but that's a minor nitpick.

Reply:
Agreed, minor issue though.

Reply:
It's honestly probably more important that the name used for the Java declaration makes sense rather than here, since that's what the user perspective will see.